source/agora/test/Base.d

```diff
@@ -853,6 +866,13 @@ public class TestValidatorNode : Validator, TestAPI
     {
         return this.qc;
     }
+
+    /// Returns an instance of a TestNominator with customizable behavior
+    protected override TestNominator getNominator ( NetworkManager network,
```

Comment:
This function is overridden so the signature should be the same as that of the superclass. But I know what you intended.

Reply:
https://en.wikipedia.org/wiki/Covariant_return_type

Reply:
There is also a very good reason to do this. There is an override of this method here: https://github.com/bpfkorea/agora/blob/97bfb80edad10731d0a5f41647428913a51dbafb/source/agora/test/EnrollmentManager.d#L383

If I didn't use `TestNominator` in `TestValidatorNode` then the code in `EnrollmentManager` could be written as:

```D
static class BadNominator : Nominator  // NOT TestNominator
{
    ...
}

protected override Nominator getNominator ( ... ) { return new BadNominator(...); }
```

This means it would circumvent the TestNominator class. But we want all derived classes to be forced to instantiate either `TestNominator` or a derived class from `TestNominator`. Changing the return type to be narrower (TestNominator in this PR line) will force the compiler to check this. It will be a compile-time error if the derived class returns an instance of a Nominator and not a TestNominator.